internal/repository/repository.go

```diff
@@ -568,6 +569,33 @@ func LoadIndex(ctx context.Context, repo restic.Repository, id restic.ID) (*Inde
 	return nil, err
 }
 
+// MasterKeyFile reads the master key from a given file
+func (r *Repository) MasterKeyFile(ctx context.Context, keyfile string) error {
+
+	buf, err := ioutil.ReadFile(keyfile) // just pass the file name
```

Comment:
[paranoia] should we check `opts.MasterKeyFile` mode and refuse to accept it unless it's `600` or `400`  like SSH does?
Maybe also warn the user that the key might have been compromised?

Reply:
[nit] not sure what is the value of the `// just pass the file name` comment. Is it needed?

Reply:
Not really sure if we should do paranoia here.. Do others have an opinion about it? If more are voting for a permission check, I'll add it.

Comment is removed - was a copy/paste error